llvm/test/CodeGen/AIE/aie2/hardware-loops/sibling.ll

```diff
@@ -45,21 +40,16 @@ define void @sibling(ptr nocapture %out, ptr nocapture readonly %in, i32 noundef
 ; CHECK-NEXT:    .p2align 4
 ; CHECK-NEXT:  .LBB0_3: // %for.body6
 ; CHECK-NEXT:    // =>This Inner Loop Header: Depth=1
-; CHECK-NEXT:    nopa ; lshl r4, r2, r3
+; CHECK-NEXT:    nopb ; nopa ; nops ; lshl r4, r2, r3; nopm ; nopv
 ; CHECK-NEXT:    mov dj0, r4
 ; CHECK-NEXT:    lda r4, [p1, dj0]
 ; CHECK-NEXT:    nop
 ; CHECK-NEXT:    nop
-; CHECK-NEXT:    nop
-; CHECK-NEXT:    nop
-; CHECK-NEXT:    nop
-; CHECK-NEXT:    add r2, r2, #1
-; CHECK-NEXT:    add r0, r0, r4
 ; CHECK-NEXT:    jnzd r1, r1, p2
 ; CHECK-NEXT:    nop // Delay Slot 5
 ; CHECK-NEXT:    nop // Delay Slot 4
-; CHECK-NEXT:    nop // Delay Slot 3
-; CHECK-NEXT:    nop // Delay Slot 2
+; CHECK-NEXT:    add r2, r2, #1 // Delay Slot 3
+; CHECK-NEXT:    add r0, r0, r4 // Delay Slot 2
```

Comment:
Checking: Those could be moved past `JNZD` because the latter also writes `srCarry` but we proved the dependency can be removed.